btree.go

```diff
@@ -895,15 +889,22 @@ func (e *Enumerator) Prev() (k interface{} /*K*/, v interface{} /*V*/, err error
 		return
 	}
 
+	if !e.hit {
+		// move to previous because Seek overshoots if there's no hit
+		if err = e.prev(); err != nil {
+			return
+		}
+	}
+
 	if e.i >= e.q.c {
-		if err = e.next(); err != nil {
+		if err = e.prev(); err != nil {
 			return
 		}
 	}
 
 	i := e.q.d[e.i]
 	k, v = i.k, i.v
-	e.k, e.hit = k, false
+	e.k, e.hit = k, true
```

Comment:
> I think I might need to change this for Next() to set e.hit = true and change the if !e.hit && hit to if e.hit && !hit as well (not sure though).

Unfortunately, I'm not sure either. The code was written too long ago. At this moment you understand it probably more than I do.

> Also, would be nice to have a test that changes the direction of iteration on enum (i.e. first Next() and then Prev()).

Agreed. Especially now when the behavior of `Prev` was fixed.